core/frontend/src/tile/RealityTileLoader.ts

```diff
@@ -143,7 +143,23 @@ export abstract class RealityTileLoader {
         reader = I3dmReader.create(streamBuffer, iModel, modelId, is3d, tile.contentRange, system, yAxisUp, tile.isLeaf, isCanceled, undefined, this.wantDeduplicatedVertices);
         break;
       case TileFormat.Gltf:
-        const props = GltfReaderProps.create(streamBuffer.nextBytes(streamBuffer.arrayBuffer.byteLength), yAxisUp);
+        /* We can obtain the base url from the tile's tree id.
+        * The id is constructed with parts from the reality data sourcekey like this - provider:format:id:itwinId
+        * The id in this case is the base url of the reality data source. We can split with ":" and parse to get the base url.
+        */
+        const treeId = tile.tree.id;
+        const split = treeId.split(":");
+
+        /* Splitting with ":" will end up splitting up our base url as well as the id.
```

Comment:
all of this seems very fragile,
please add unit test for these

Reply:
@aruniverse does the added unit test suffice?